src/components/Quotes.jsx

```diff
@@ -0,0 +1,68 @@
+import './Quotes.css';
+import { useState } from 'react';
+
+function Quotes() {
+  const [quote, setQuote] = useState('');
+  const [author, setAuthor] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const category = 'success';
+
+  const getQuotes = async () => {
+    const url = `https://api.api-ninjas.com/v1/quotes?category=${category}`;
+    try {
+      setIsLoading(true);
+      fetch(url, {
+        method: 'GET',
+        withCredentials: true,
+        headers: {
+          'X-Api-Key': 'c5cFywTgJ2lZOt0YMl8FPw==e4WI89mcjpo3tZZW',
+          'Content-Type': 'application/json',
+        },
+      })
+        .then((response) => response.json())
+        .then((data) => {
+          setQuote(data[0].quote);
+          setAuthor(data[0].author);
+        });
+      setIsLoading(false);
+    } catch (error) {
+      setIsLoading(false);
+      setErrorMessage(error);
+    }
+  };
```

Comment:
Nice job so far fetching the quotes😉. But for this milestone, we are required to make use of the useEffect hook when making the `http` request😉.
<img width="474" alt="image" src="https://github.com/fickryiman/math-magicians-rep/assets/64643995/835f4e09-533f-4033-b872-2a39dde6a4f4">

Kindly note that the useEffect hook and the useState hook serve different purposes in React applications. While the useState hook is primarily used for managing the state within a component, the useEffect hook is specifically designed to handle side effects, such as making HTTP requests😉.

You might want to consider using the useEffect hook because the useEffect hook allows you to perform side effects after React has rendered the component, ensuring that the DOM is ready. Making an HTTP request is an asynchronous operation, and it's important to initiate it at the appropriate time in the component's lifecycle. Placing the HTTP request within a `useEffect` hook ensures that it's executed at the right moment😉. also, If you make an HTTP request directly inside the component body or within a `useState` setter function, it can potentially lead to an infinite loop. When the component re-renders due to a state change caused by the HTTP response, it triggers the HTTP request again, resulting in an endless loop. useEffect allows you to control when the effect should be executed, preventing such loops. I can see you tried to manage that by using a click event but we might just want to do it the right way by using the hook created for it😉.